dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -31,39 +31,47 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.time.LocalDateTime;
+import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
 // TODO this is leaking aether too far
 import org.eclipse.aether.collection.DependencyCollectionException;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.resolution.ArtifactDescriptorException;
+import org.eclipse.aether.resolution.ArtifactDescriptorRequest;
+import org.eclipse.aether.resolution.ArtifactDescriptorResult;
 import org.eclipse.aether.resolution.DependencyResolutionException;
 
 import com.google.cloud.tools.opensource.dependencies.DependencyGraph;
 import com.google.cloud.tools.opensource.dependencies.DependencyGraphBuilder;
+import com.google.cloud.tools.opensource.dependencies.RepositoryUtility;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Splitter;
 
 public class DashboardMain {
   
-  // todo move this to config file
-  static String[] ARTIFACTS = {
-      "com.google.cloud:google-cloud-core:1.41.0",
-      "com.google.cloud:google-cloud-datastore:1.41.0"
-  };
-  
   public static void main(String[] args) 
-      throws IOException, TemplateException {
+      throws IOException, TemplateException, ArtifactDescriptorException {
```

Comment:
It doesn't seem useful to declare explicit exceptions that `main` throws - why not use use `throws Exception`?

Reply:
I'm on the far end of the checked exceptions distribution. Declare all the things! If I accidentally introduce code that throws a new exception type, I want a red squiggly line so I can think about what I've done and whether I really want that.